pkg/signup/keymanager.go

```diff
@@ -0,0 +1,172 @@
+package signup
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"gopkg.in/square/go-jose.v2"
+)
+
+// PublicKey represents an RSA public key with a Key ID
+type PublicKey struct {
+	KeyID string
+	Key   *rsa.PublicKey
+}
+
+// JSONKeys the remote keys encoded in a json document
+type JSONKeys struct {
+	Keys []interface{} `json:"keys"`
+}
+
+// KeyManager manages the public keys for token validation.
+type KeyManager struct {
+	config *configuration.Registry
+	logger *log.Logger
```

Comment:
why do you need a pointer to the logger?

Reply:
What is the alternative? We're always handing around pointers to the logger there as the logger is created with some params in the server struct..

Reply:
ok, I guess it makes it "shared" amongst services, which may be better if it manages file descriptors..